### PR TITLE
feature/separate-index-per-env

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a user I do not want my data to leak across environments

## Changes proposed in this pull request

The `ElasticsearchStorageHandler` no longer has a default `root_index` but now fetches it from the `Settings`. this enables us to set different root_indexes per environment

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/9300602769

### CODE REVIEW

File: infrastructure/aws/ecs.tf [hunk 1]
```diff
@@ -5,6 +5,7 @@ locals {
   environment_variables = {
     "ELASTIC__API_KEY" : var.elastic_api_key,
     "ELASTIC__CLOUD_ID" : var.cloud_id,
+    "ROOT_INDEX": "redbox-date-${terraform.workspace}",
```
Comment (252afh) [inline infrastructure/aws/ecs.tf:8]: Typo in `date` -> `data`

File: redbox/models/settings.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ class Settings(BaseSettings):
     partition_strategy: Literal["auto", "fast", "ocr_only", "hi_res"] = "fast"
 
     elastic: ElasticCloudSettings | ElasticLocalSettings = ElasticLocalSettings()
+    root_index: str = "redbox-data"
```
Comment (252afh) [inline redbox/models/settings.py:47]: Do we want this to be the same as what's in the integration .env file?

File: .env.integration [hunk 3]
```diff
@@ -5,3 +5,4 @@ COMPRESSION_ENABLED=False
 AWS_REGION=eu-west-2
 DJANGO_LOG_LEVEL=WARNING
 OPENAI_MODEL=azure/gpt-35-turbo-16k
+ROOT_INDEX=redbox-data-integration
```
Comment (252afh) [inline .env.integration:8]: Could we rename this to `ELASTIC_ROOT_INDEX` to be more descriptive, please?

**Review by 252afh — APPROVED** [review]

File: redbox/models/settings.py (hunk 2 above)
Comment (gecBurton) [reply redbox/models/settings.py:47]: nah, just some sensible default i think?

File: infrastructure/aws/ecs.tf (hunk 1 above)
Comment (gecBurton) [reply infrastructure/aws/ecs.tf]: [eb53e0d](https://github.com/i-dot-ai/redbox-copilot/pull/482/commits/eb53e0dbc6604b3d591eaaf23268b4d84131ee7e)

File: .env.integration (hunk 3 above)
Comment (gecBurton) [reply .env.integration]: [7c6003e](https://github.com/i-dot-ai/redbox-copilot/pull/482/commits/7c6003e777f0ad6915c13838b5db12b5e4cb3acc)

**Pull request merged**